zerver/views/user_groups.py

```diff
@@ -35,6 +35,10 @@ def edit_user_group(request: HttpRequest, user_profile: UserProfile,
         return json_error(_("No new data supplied"))
 
     user_group = access_user_group_by_id(user_group_id, realm=user_profile.realm)
+    group_member_ids = get_user_group_members(user_group)
+    msg = _("Only user group members and organization administrators are allowed to edit groups")
+    if (not user_profile.is_realm_admin and user_profile.id not in group_member_ids):
+        return json_error(msg)
```

Comment:
This code block is duplicated in several places.  You should put it into a function, and have it raise `JsonableError` (an exception which results in an equivalent `json_error`) 

Or, better, we should refactor `access_user_group_by_id` to accept a user object rather than a realm